### DISCUSSION TITLE
CSI VolumeSnapshot is not working in K8s to take backup for PVC

### DISCUSSION THREAD

**Discussion by ibrahimhalilkoyuncu** [longhorn/longhorn#5652] [Q&A] [answered]
Hello everyone,

I am trying to take VolumeSnapshot to backup purpose and when try to create volumesnapshot, volumesnapshotcontent object "readyToUse" flag is not set with "True" so I can not take a snapshot from existing PVC. 

Longhorn used as storage class and version is 1.3.2.
K8s version : 1.24.4

csi-snapshooter logs:
![image](https://user-images.githubusercontent.com/40371684/227228921-e3ea0ee5-a6c3-48ba-be0b-de567b97bc08.png)

csi-driver is below
![image](https://user-images.githubusercontent.com/40371684/227208547-49bbdc2c-3090-4972-aaab-296a40dc5bff.png)

VolumeSnapshot is below
![image](https://user-images.githubusercontent.com/40371684/227209152-bca0bb84-3b23-42b5-b880-37680fd6ded8.png)

VolumeSnapshotClass
![image](https://user-images.githubusercontent.com/40371684/227209388-2cb40f72-cd1c-4cdc-9a47-5ec2baa8bf98.png)

VolumeSnapshotContent
![image](https://user-images.githubusercontent.com/40371684/227209780-846fcf40-8eca-4e77-85e2-a6c50594685b.png)

Snapshot Controller Logs:
![image](https://user-images.githubusercontent.com/40371684/227210465-0416e49e-98ad-4adc-8daf-d69ef9fbf31c.png)

Is anyone encounter with this problem ?

**Comment by mantissahz** [comment]
Hi @menendes,
How did you [enable CSI snapshot support ](https://longhorn.io/docs/1.3.2/snapshots-and-backups/csi-snapshot-support/enable-csi-snapshot-support/)for Longhorn v1.3.2 (Could you provide your installation steps) ?
Could you provide the information by this command "kubectl describe volumesnapshotcontent snapcontent-2f849850-xxxx"?
And also check if there is any abnormal log in the Longhorn csi-snapshotter.

**Reply by ibrahimhalilkoyuncu** [reply]
I follow the the [enable CSI snapshot support docs.](https://longhorn.io/docs/1.3.2/snapshots-and-backups/csi-snapshot-support/enable-csi-snapshot-support/) But my problem resolved when I upgrade the longhorn version 1.3.2 to 1.4.1. But current problem related with pvc creation from volume snapshot. 
When I tried to create pvc it stuck in pending status and describe logs says that "failed to verify data source: size of target volume is different than size of source volume. But I specify new pvc size greater than old one which is taken snapshot. Related logs in the below. Do you have any idea ? @mantissahz

![image](https://user-images.githubusercontent.com/40371684/227444254-6b58611a-ed49-4596-aad7-cc3947b256c3.png)

**Reply by mantissahz** [reply, marked as answer]
```
Note that the spec.resources.requests.storage value must be the same as the size of VolumeSnapshot object
```

@menendes  Size must be the same.
